internal/cache/file/downloader/job.go

```diff
@@ -341,10 +318,6 @@ func (job *Job) downloadObjectAsync() {
 							ReadCompressed: job.object.HasContentEncodingGzip(),
 						})
 					if err != nil {
-						if errors.Is(err, context.Canceled) {
```

Comment:
Just thinking, if someone will call invalidate anyway we would be removing the job. Hence, we are not handling here?

Reply:
We have removed the cancellation method from Job so cancellation should not happen and if it ever happens for any reason it should be reported as failure.

Reply:
When cache is invalidated at that time `job.cancel` is called. Right? And ideally that's not a failure case. 
Or let me know, if I am missing something.

Reply:
There is no harm in that:
1. The error in failure will say context cancelled.
2. When the job is Invalidated or Failed, the behavior is same for caller i.e. to not read from cache. (earlier it was not because the caller can read from Cancelled state).

If you still think it creates confusion logically then we can notify with Invalid instead of Failure.

Reply:
Done, notify with Invalid in case context is canceled to avoid giving logger.Errorf in logs (that comes in notify failure) which can create little confusion.